app/lib/tenejo/preflight.rb

```diff
@@ -136,9 +136,18 @@ def self.map_header(m)
       KNOWN_HEADERS.find { |x| x.to_s.gsub(/[^0-9A-Za-z]/, '') == squashed } || m
     end
 
-    def self.read_csv(input, import_path)
+    def self.read_csv(filename, import_path = DEFAULT_UPLOAD_PATH)
+      stream = File.open(filename)
       begin
-        csv = CSV.open(input, headers: true, return_headers: true, skip_blanks: true,
+        process_csv(stream, import_path)
+      ensure
+        stream.close
+      end
+    end
+
```

Comment:
Active Storage makes it easy to get the contents of the uploaded CSV back into a string, it's actually harder to get them back into a temp file.  So I created an entry point into the preflight processing that allowed me to supply a string with the full CSV instead of a filename.  There might be a better way to do this, or we might want to just assume that the input to the processor will always be an in-memory string - that seems like it might match the expectations of active storage better.